### PR TITLE
fix(shared): preserve unicode group labels in slug normalization

### PR DESCRIPTION
**Pull request by koen666**
# fix(shared): preserve Unicode group names in slug normalization

Group display names with non-Latin scripts were being stripped during slug normalization, which made session subtitles collapse to provider-only labels (for example, only `telegram`).

Fixes #58932

## Summary

- **Problem:** `normalizeHyphenSlug` and `normalizeAtHashSlug` used ASCII-only regex and removed CJK/Cyrillic/Arabic characters
- **Why it matters:** group session labels become indistinguishable when names are non-Latin
- **What changed:** switched to Unicode-aware character classes (`\p{L}` / `\p{N}`) and added regression tests for non-Latin scripts
- **What did NOT change (scope boundary):** no changes to session routing, metadata pipeline, provider logic, or UI rendering

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [ ] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [ ] API / contracts
- [x] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #58932
- Related # N/A
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

- **Root cause:** slug normalization regex only allowed `[a-z0-9...]`, so Unicode letters/numbers were replaced and often collapsed away
- **Missing detection / guardrail:** tests covered ASCII behavior but not multilingual group names
- **Prior context:** issue #58932 reported session subtitle degradation for non-Latin group names
- **Why this regressed now:** not a new regression; this appears to be a long-standing behavior gap

## Regression Test Plan (if applicable)

- Coverage level that should have caught this:
  - [x] Unit test
  - [ ] Seam / integration test
  - [ ] End-to-end test
  - [ ] Existing coverage already sufficient
- **Target test or file:** `src/shared/string-normalization.test.ts`
- **Scenario the test should lock in:** CJK/Cyrillic/Arabic characters are preserved in slug normalization paths
- **Why this is the smallest reliable guardrail:** the bug is in shared normalization helpers; direct unit tests pin exact behavior

## User-visible / Behavior Changes

- Group display names with non-Latin scripts are now preserved in normalized labels
- Sessions list can distinguish multilingual groups on the same provider

## Diagram (if applicable)

```text
Before:
"技术讨论组" -> normalizeHyphenSlug -> "" -> fallback display "telegram"

After:
"技术讨论组" -> normalizeHyphenSlug -> "技术讨论组" -> display keeps group name
```

## Security Impact (required)

- New permissions/capabilities? `No`
- Secrets/tokens handling changed? `No`
- New/changed network calls? `No`
- Command/tool execution surface changed? `No`
- Data access scope changed? `No`

### Environment

- OS: macOS
- Runtime/container: Node 22+, pnpm workspace
- Model/provider: N/A
- Integration/channel (if any): N/A
- Relevant config (redacted): N/A

### Steps

1. Run `pnpm test -- src/shared/string-normalization.test.ts`
2. Verify Unicode test cases pass for CJK/Cyrillic/Arabic inputs

### Expected

- Non-Latin group names remain in normalized output

### Actual

- After fix, normalization preserves Unicode letters/numbers and tests pass

## Evidence

- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

```bash
pnpm test -- src/shared/string-normalization.test.ts
# Test Files  1 passed (1)
# Tests       9 passed (9)
```

## Human Verification (required)

- **Verified scenarios:**
  - `normalizeHyphenSlug` preserves Chinese/Japanese/Korean/Cyrillic/Arabic strings
  - `normalizeAtHashSlug` preserves CJK strings with `#`/`@` prefixes
  - existing ASCII normalization expectations remain green
- **What you did NOT verify:**
  - full end-to-end session subtitle rendering in a live multi-channel environment

## Compatibility / Migration

- Backward compatible? `Yes`
- Config/env changes? `No`
- Migration needed? `No`

## Risks and Mitigations

- **Risk:** broader Unicode acceptance may retain more characters than before in edge-case labels
  - **Mitigation:** normalization still collapses unsupported separators and trims boundary punctuation; tests lock intended behavior

### CODE REVIEW

File: src/shared/string-normalization.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export function normalizeHyphenSlug(raw?: string | null) {
     return "";
   }
   const dashed = trimmed.replace(/\s+/g, "-");
-  const cleaned = dashed.replace(/[^a-z0-9#@._+-]+/g, "-");
+  const cleaned = dashed.replace(/[^\p{L}\p{N}#@._+-]+/gu, "-");
   return cleaned.replace(/-{2,}/g, "-").replace(/^[-.]+|[-.]+$/g, "");
```
[the remark this answers is not in the document]
Comment (frankentini) [reply src/shared/string-normalization.ts]: Addressed in commit 02c7ea66e8 — added `\p{M}` to the character class and NFC normalization before processing. Decomposed forms now normalize consistently.

File: src/shared/string-normalization.ts [hunk 2]
```diff
@@ -23,6 +23,6 @@ export function normalizeAtHashSlug(raw?: string | null) {
   }
   const withoutPrefix = trimmed.replace(/^[@#]+/, "");
   const dashed = withoutPrefix.replace(/[\s_]+/g, "-");
-  const cleaned = dashed.replace(/[^a-z0-9-]+/g, "-");
+  const cleaned = dashed.replace(/[^\p{L}\p{N}-]+/gu, "-");
   return cleaned.replace(/-{2,}/g, "-").replace(/^-+|-+$/g, "");
```
[the remark this answers is not in the document]
Comment (frankentini) [reply src/shared/string-normalization.ts]: Addressed in same commit — `normalizeAtHashSlug` now also includes `\p{M}` and NFC normalization.

File: src/shared/string-normalization.test.ts [hunk 3]
```diff
@@ -45,4 +45,34 @@ describe("shared/string-normalization", () => {
...
+    expect(normalizeHyphenSlug("Team 技术讨论")).toBe("team-技术讨论");
+    expect(normalizeHyphenSlug("#OpenClaw中文群")).toBe("#openclaw中文群");
+  });
+
+  it("preserves Cyrillic and Arabic characters in normalizeHyphenSlug", () => {
+    // Cyrillic (Russian)
+    expect(normalizeHyphenSlug("Команда разработки")).toBe("команда-разработки");
+    // Arabic
+    expect(normalizeHyphenSlug("فريق التطوير")).toBe("فريق-التطوير");
+  });
```
[the remark this answers is not in the document]
Comment (frankentini) [reply src/shared/string-normalization.test.ts]: Added test cases for decomposed accented characters (e\u0301cole → école) and Devanagari with combining marks (हिन्दी, Thai สวัสดี).

File: src/shared/string-normalization.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ export function normalizeHyphenSlug(raw?: string | null) {
     return "";
   }
   const dashed = trimmed.replace(/\s+/g, "-");
-  const cleaned = dashed.replace(/[^a-z0-9#@._+-]+/g, "-");
+  const cleaned = dashed.replace(/[^\p{L}\p{N}#@._+-]+/gu, "-");
```
[the remark this answers is not in the document]
Comment (frankentini) [reply src/shared/string-normalization.ts]: Addressed — `\p{M}` is now included in both slug normalizers, and inputs are NFC-normalized first. Hindi and decomposed Latin test cases added to verify.

**Comment by vincentkoc** [comment]
ProjectClownfish pushed a narrow repair to this branch so the original contributor path can stay canonical.

Source PR: https://github.com/openclaw/openclaw/pull/59068
Validation: pnpm -s vitest run src/shared/string-normalization.test.ts; pnpm check:changed
Contributor credit is preserved in the branch history and PR context.
